File: playground/examples/node/retry.js
const { task, flow, runWorkflow } = require("sayiir");

const fetchData = task("fetch-data", (query) => {
  return `Result for: ${query}`;
}, { timeout: "5s", retries: 3 });

const workflow = flow("search").then(fetchData).build();
runWorkflow(workflow, "sayiir").then(console.log);

File: playground/examples/python/retry.py
from sayiir import task, Flow, run_workflow

@task(timeout="5s", retries=3)
def fetch_data(query: str) -> str:
    return f"Result for: {query}"

workflow = Flow("search").then(fetch_data).build()
print(run_workflow(workflow, "sayiir"))
